qemu/tests/smbios_table.py

```diff
@@ -59,7 +64,7 @@ def run(test, params, env):
 
     failures = []
     for m_type in support_machine_types:
-        params["machine_type"] = m_type
+        params["machine_type"] = "%s%s" % (machine_type_prefix, m_type)
```

Comment:
This change only work with "traversal_machine_emulated = yes".
If "traversal_machine_emulated = no", then "support_machine_types" equal to "arm64-pci:virt" (for example), it contains "arm64-pci:", and then machine_type will be "arm64-pci:arm64-pci:virt".  

In qcontainer.py, will cause arm_machine= arm64-pci, machine_type=arm64-pci:virt,  cause failed as " Unsupported machine type arm64-pci:virt"
line 995   if machine_type.startswith('arm64'):                                
line 996       arm_machine, machine_type = machine_type.split(':', 1)